### DISCUSSION TITLE
HTML links don't seem to be convert in DOCX documents

### DISCUSSION THREAD

**Discussion by ptram** [quarto-dev/quarto-cli#9247] [Q&A]
### Description

Hi,

It looks like HTML links are not converted when creating DOCX documents.

I'm not trying to use the basic Markdown syntax, that is:

`[shown_item](www.linked_website.org)`

but the HTML syntax, like this:

`<a href="www.linked_website.org">shown_item</a>`

I want to do this, because I need to open the link into a new window, using the target="_blank" argument in the tag. This doesn't look to be possible with Markdown links.

However, it seems that the HTML link is not converted to a link into the DOCX document. The link syntax disappear, but then the `shown_item` word can't be clicked as a hyperlink (and doesn't appear as one).

**Comment by mcanouil** [comment]
Could you share a small **self-contained "working" (reproducible)** example to work with rather than small snippet of code which are missing the context, _i.e._, a complete Quarto document or a Git repository? Thanks.

Snippet/inline code cannot be copy/pasted for instance in addition to the missing context.


---

You can share a **self-contained "working" (reproducible)** Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).
See <https://quarto.org/bug-reports.html#small-is-beautiful-aim-for-a-single-document-with-10-lines>.

If you have multiple files (and if it is **absolutely required** to have multiple files), please share as a Git repository.

<table>
<tr><th>R</th><th>Python</th></tr>
<tr><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: knitr
---

This is a reproducible Quarto document.

```{r}
x <- c(1, 2, 3, 4, 5)
y <- c(1, 4, 9, 16, 25)

plot(x, y)
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td><td>

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
engine: jupyter
---

This is a reproducible Quarto document.

```{python}
import matplotlib.pyplot as plt

x = [1, 2, 3, 4, 5]
y = [1, 4, 9, 16, 25]

plt.plot(x, y)
plt.show()
```

![An image](https://placehold.co/600x400.png)

The end.
````
`````

</td></tr>
</table>

Additionally and if not already given, please share the output of `quarto check` within a code blocks (*i.e.*, using three backticks ` ```txt `), see <https://quarto.org/bug-reports.html#check>.

**Reply by mcanouil** [reply]
Now, in the documentation, there is an example showing how to add attributes using brackets, see <https://quarto.org/docs/authoring/markdown-basics.html#links-images> and how to add raw content, see <https://quarto.org/docs/authoring/markdown-basics.html#raw-content>

````qmd
---
format: html
---

[Quarto](https://quarto.org){target=_blank}

`<a href="https://quarto.org" target="_blank">Quarto</a>`{=html}
````

**Comment by cscheid** [comment]
```
$ pandoc -f markdown -t html
[a link](https://example.com){target=_blank}
^D
<p><a href="https://example.com" target="_blank">a link</a></p>
```

**Reply by ptram** [reply]
@cscheid and @mcanouil , thank you for your hints. The Markdown version you suggested worked great, and is much easier to manage. Sorry to have missed it in the manual. Have been investigating for it a few weeks ago, but obviously not in the right place.

The HTML version that is not working can be seen in the following snippet:

````
---
title: A hyperlink example
author: Me Myself
format:
  docx:
---
  
# This is the title

And this is the text, with a non-working reference to <a href="https://quarto.org/" target="_blank">Quarto</a>.

This one works great: [Quarto](https://quarto.org/){target=_blank}.
````

**Reply by mcanouil** [reply]
As showed, you should not write raw code outside of raw markup.
Microsoft Word uses OpenXML which is close to HTML but it is not HTML. You'll hit the exact same issue if you use raw HTML for LaTeX or Typst, and the other way around. There is nothing Quarto can really do here.

Quarto main input is Markdown. If you use something else, then it means you know what you are doing.
